src/algebra/hom/group.lean

```diff
@@ -1088,20 +1088,20 @@ For the iff statement on the triviality of the kernel, see `monoid_hom.injective
 @[to_additive "A homomorphism from an additive group to an additive monoid is injective iff
 its kernel is trivial. For the iff statement on the triviality of the kernel,
 see `add_monoid_hom.injective_iff'`."]
-lemma injective_iff {G H} [group G] [mul_one_class H] (f : G →* H) :
+lemma injective_iff {G H} [group G] [mul_one_class H] [monoid_hom_class F G H] (f : F) :
```

Comment:
Maybe worth renaming to
```suggestion
lemma _root_.injective_iff_map_eq_one {G H} [group G] [mul_one_class H] [monoid_hom_class F G H] (f : F) :
```
Since the namespace is no longer accurate